modules/core/src/main/scala/doobie/hi/connection.scala

```diff
@@ -543,4 +577,24 @@ object connection {
   // val nativeTypeMap: ConnectionIO[Map[String, JdbcType]] = {
   //   getMetaData(IFDMD.getTypeInfo.flatMap(IFDMD.embed(_, HRS.list[(String, JdbcType)].map(_.toMap))))
   // }
+
+  final case class PreparedExecutionWithResultSet[A](
+      create: ConnectionIO[PreparedStatement],
+      prep: PreparedStatementIO[Unit],
+      exec: PreparedStatementIO[ResultSet],
+      process: ResultSetIO[A]
+  )
+
+  final case class PreparedExecutionWithoutResultSet[A](
+      create: ConnectionIO[PreparedStatement],
+      prep: PreparedStatementIO[Unit],
+      exec: PreparedStatementIO[A]
+  )
+
+  final case class PreparedExecutionStream(
+      create: ConnectionIO[PreparedStatement],
+      prep: PreparedStatementIO[Unit],
+      exec: PreparedStatementIO[ResultSet],
+      chunkSize: Int
+  )
```

Comment:
If we remove `chunkSize` then this will be the same as `PreparedExecutionWithoutResultSet[ResultSet]`. That does look weird though, so I think we can rename `PreparedExecutionWithoutResultSet` to something like `PreparedExecutionWithoutProcessStep`? :smile: